src/extensions/updateHelper.ts

```diff
@@ -244,6 +167,255 @@ export interface UpdateOptions {
+    existingSpec,
+    existingSource
+  );
+}
+
+/**
+ * Preparatory work for updating an published extension instance to the given version.
+ *
+ * @param instanceId Id of the instance to update
+ * @param existingSpec ExtensionSpec of the existing instance source
+ * @param existingSource name of existing instance source
+ * @param version Version to update the instance to
+ */
+export async function updateToVersionFromRegistry(
+  projectId: string,
+  instanceId: string,
+  existingSpec: extensionsApi.ExtensionSpec,
+  existingSource: string,
+  version: string
+): Promise<string> {
+  if (version !== "latest" && !semver.valid(version)) {
+    throw new FirebaseError(`cannot update to invalid version ${version}`);
+  }
+  // Updating to a version from an published source
+  let registryEntry;
+  try {
+    registryEntry = await resolveSource.resolveRegistryEntry(existingSpec.name);
+  } catch (err) {
+    // If registry entry does not exist, assume existing source was from local directory or URL.
+    throw new FirebaseError(
+      `Cannot find the latest version of this extension. To update this instance to a local source or URL source, run "firebase ext:update ${instanceId} <localSourceOrURL>".`
+    );
+  }
+  utils.logLabeledBullet(
+    logPrefix,
+    clc.bold("You are updating this extension instance to an official source.")
+  );
+
+  // Do not allow user to "downgrade" to a version lower than the minimum required version.
+  const minVer = resolveSource.getMinRequiredVersion(registryEntry);
+  if (minVer) {
+    if (version !== "latest" && semver.gt(minVer, version)) {
+      throw new FirebaseError(
+        `The version you are trying to upgrade to (${clc.bold(
+          version
+        )}) is less than the minimum version required (${clc.bold(minVer)}) to use this extension.`
+      );
+    }
+  }
+  const targetVersion = resolveSource.getTargetVersion(registryEntry, version);
+  await showUpdateVersionInfo(instanceId, existingSpec.version, targetVersion);
+  const warning =
+    "All the instance's extension-specific resources and logic will be overwritten to use the source code and files from the latest released version.\n\n";
+  const existingSourceOrigin = await getExistingSourceOrigin(
+    projectId,
+    instanceId,
+    existingSpec.name,
+    existingSource
+  );
+  await module.exports.warningUpdateToOtherSource(
+    existingSourceOrigin,
+    warning,
+    SourceOrigin.OFFICIAL_EXTENSION
+  );
+  await resolveSource.promptForUpdateWarnings(registryEntry, existingSpec.version, targetVersion);
+  return resolveSource.resolveSourceUrl(registryEntry, existingSpec.name, targetVersion);
+}
+
+/**
+ * Preparatory work for updating an published extension instance to the latest version.
+ *
+ * @param instanceId Id of the instance to update
+ * @param existingSpec ExtensionSpec of the existing instance source
+ * @param existingSource name of existing instance source
+ */
+export async function updateFromRegistry(
```

Comment:
`async` is not necessary here since you're not `await`-ing anything

Reply:
Hmmm... Not sure I follow on this one. 

updateFromRegistry returns the output of calling updateToVersionFromRegistry(), which returns a promise. 

updateToVersionFromRegistry() is an async function that needs to await an API call. So as far as I understand, this is still needed?